app/controllers/home_controller.rb

```diff
@@ -1,7 +1,38 @@
 class HomeController < ApplicationController
-  skip_before_action :authenticate_user!
+  skip_before_action :authenticate_user!, except: :dashboard
+  respond_to :html
 
   def index
     redirect_to time_entries_url if user_signed_in?
   end
+
+  def dashboard
+    ## Main objects
+    @admin = current_user.admin?
+    @date = params[:date] ? Date.parse(params[:date]) : Date.today
+    @tasks = Task.includes(:tags)
+    @time_entries = TimeEntry.all
+    @tags = @admin ? Tag.all : Tag.where(user: current_user)
```

Comment:
While this ternary syntax is more concise, I think I prefer the syntax you used for `@time_entries` and `@tasks` better. To consolidate, we could change this section to
```ruby
@tasks = Task.includes(:tags)
@time_entries = TimeEntry.all
@tags = Tag.all
```

And then below have:
```ruby
# Limit to just ours if we're not an admin
unless @admin
  @tasks = @tasks.where(user: current_user)
  @time_entries = @time_entries.where(user: current_user)
  @tags.where(user: current_user)
end